hw3/02-charts.js

```diff
@@ -1,52 +1,296 @@
+						hashMap.set("Tarth", 1);
+					}
+				} else if (family[i].includes("Naathi")) {
+					if (hashMap.has("Naathi")) {
+						hashMap.set("Naathi", hashMap.get("Naathi") + 1);
+					} else {
+						hashMap.set("Naathi", 1);
+					}
+				} else if (family[i].includes("Bolton")) {
+					if (hashMap.has("Bolton")) {
+						hashMap.set("Bolton", hashMap.get("Bolton") + 1);
+					} else {
+						hashMap.set("Bolton", 1);
+					}
+				} else if (family[i].includes("Naharis")) {
+					if (hashMap.has("Naharis")) {
+						hashMap.set("Naharis", hashMap.get("Naharis") + 1);
+					} else {
+						hashMap.set("Naharis", 1);
+					}
+				} else if (
+					family[i].includes("Lorathi") ||
+					family[i].includes("Lorath")
+				) {
+					if (hashMap.has("Lorath")) {
+						hashMap.set("Lorath", hashMap.get("Lorath") + 1);
+					} else {
+						hashMap.set("Lorath", 1);
+					}
+				} else if (family[i].includes("Sand")) {
+					if (hashMap.has("Sand")) {
+						hashMap.set("Sand", hashMap.get("Sand") + 1);
+					} else {
+						hashMap.set("Sand", 1);
+					}
+				} else if (family[i].includes("Worm")) {
+					if (hashMap.has("Worm")) {
+						hashMap.set("Worm", hashMap.get("Worm") + 1);
+					} else {
+						hashMap.set("Worm", 1);
+					}
+				} else if (family[i].includes("Qyburn")) {
+					if (hashMap.has("Qyburn")) {
+						hashMap.set("Qyburn", hashMap.get("Qyburn") + 1);
+					} else {
+						hashMap.set("Qyburn", 1);
+					}
+				} else if (family[i].includes("Bronn")) {
+					if (hashMap.has("Bronn")) {
+						hashMap.set("Bronn", hashMap.get("Bronn") + 1);
+					} else {
+						hashMap.set("Bronn", 1);
+					}
+				} else if (family[i].includes("Sparrow")) {
+					if (hashMap.has("Sparrow")) {
+						hashMap.set("Sparrow", hashMap.get("Sparrow") + 1);
+					} else {
+						hashMap.set("Sparrow", 1);
+					}
+				} else if (family[i].includes("Mormont")) {
+					if (hashMap.has("Mormont")) {
+						hashMap.set("Mormont", hashMap.get("Mormont") + 1);
+					} else {
+						hashMap.set("Mormont", 1);
+					}
+				} else if (family[i].includes("Viper")) {
+					if (hashMap.has("Viper")) {
+						hashMap.set("Viper", hashMap.get("Viper") + 1);
+					} else {
+						hashMap.set("Viper", 1);
+					}
+				} else if (
+					family[i].includes("Unknown") ||
+					family[i].includes("None") ||
+					family[i].includes("Unkown") ||
+					family[i].includes("")
+				) {
+					if (hashMap.has("Unknown")) {
+						hashMap.set("Unknown", hashMap.get("Unknown") + 1);
+					} else {
+						hashMap.set("Unknown", 1);
+					}
+				}
+			}
+
+			let labels = [];
+			let values = [];
+
+			for (let [key, value] of hashMap) {
+				labels.push(key);
+				values.push(value);
+			}
+			renderChart(labels, values);
+}
+
+const renderChart = (labels, values) => {
+	const donutChart = document.querySelector(".donut-chart");
+	new Chart(donutChart, {
+		type: "doughnut",
+		data: {
+			labels: [...labels],
+			datasets: [
+				{
+					label: "My First Dataset",
```

Comment:
This label field is for the 'units' for describing the data in the little pop up that describe the data values. So 'People' or '# of members' would work here.